### PR TITLE
[Improve] Extend `SupportResourceShare` to spark/flink

### PR DESCRIPTION
**Pull request by Hisoka-X**
### Purpose of this pull request
This PR extend `SupportResourceShare` to spark/flink

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
add new test.

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by Hisoka-X** [comment]
waiting https://github.com/apache/seatunnel/pull/6116

**Comment by Hisoka-X** [comment]
cc @TyrantLucifer @hailin0

**Review by EricJoy2048 — APPROVED** [review]

File: seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/multitablesink/MultiTableSinkAggregatedCommitter.java [hunk 1]
```diff
@@ -58,16 +57,15 @@ private void initResourceManager() {
...
+                            .initMultiTableResourceManager(aggCommitters.size(), 1);
             break;
         }
         if (resourceManager != null) {
             for (String tableIdentifier : aggCommitters.keySet()) {
                 SinkAggregatedCommitter<?, ?> aggCommitter = aggCommitters.get(tableIdentifier);
                 aggCommitter.init();
                 ((SupportMultiTableSinkAggregatedCommitter<?>) aggCommitter)
-                        .setMultiTableResourceManager(Optional.of(resourceManager), 1);
+                        .setMultiTableResourceManager(resourceManager, 0);
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/multitablesink/MultiTableSinkAggregatedCommitter.java:68]: why update 1 - > 0 ?

Comment (Hisoka-X) [reply seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/multitablesink/MultiTableSinkAggregatedCommitter.java:68]: It is index of committer, so it should be started with 0.

**Review by hailin0 — APPROVED** [review]

**Pull request merged**